packages/language-server/src/lib/documents/utils.ts

```diff
@@ -215,3 +215,33 @@ export function isInTag(position: Position, tagInfo: TagInformation | null): boo
 export function getTextInRange(range: Range, text: string) {
     return text.substring(offsetAt(range.start, text), offsetAt(range.end, text));
 }
+
+export function getVariableAtPosition(position: Position, text: string) {
+    return getVariableAtOffset(offsetAt(position, text), text);
+}
+
+export function getVariableAtOffset(offset: number, text: string) {
```

Comment:
also variable.b, variable) and operators without space in between.

Reply:
I found a hack that can be used to find the identifier/symbol in the generated tsx:
```ts
const offset = fragment.offsetAt(fragment.getGeneratedPosition(position));
const { start, length } = lang.getSmartSelectionRange(tsDoc.filePath, offset).textSpan;
const identifier = tsDoc.getText(start, start + length);
```